core-db/src/main/java/io/novafoundation/nova/core_db/dao/MetaAccountDao.kt

```diff
@@ -188,6 +188,23 @@ interface MetaAccountDao {
 
     @Query("UPDATE meta_accounts SET status = :status WHERE id IN (:metaIds)")
     suspend fun changeAccountsStatus(metaIds: List<Long>, status: MetaAccountLocal.Status)
+
+    @Query(
+        """
+        WITH RECURSIVE Sequence AS (
+            SELECT pa.proxyMetaId
+            FROM proxy_accounts pa
+            WHERE pa.proxiedMetaId = :proxiedMetaId
+            UNION ALL
+            SELECT pa.proxyMetaId
+            FROM proxy_accounts pa
+            INNER JOIN Sequence s ON s.proxyMetaId = pa.proxiedMetaId
+        )
+        SELECT MAX(proxyMetaId) as finalMetaId
+        FROM Sequence
+    """
+    )
+    fun getLastProxyAccountId(proxiedMetaId: Long): Long?
```

Comment:
This recursion find root proxy account in all chain. It works nice for non-nested proxy-accounts, but I didn't check how it works for nested, I guess it would be also good